test/Fields/reduction_cuda_distributed.jl

```diff
@@ -0,0 +1,198 @@
+using Test
+using CUDA
+using ClimaComms
+using Statistics
+using LinearAlgebra
+using Logging
+
+import ClimaCore:
+    Domains,
+    Fields,
+    Geometry,
+    Meshes,
+    Operators,
+    Spaces,
+    Topologies,
+    DataLayouts
+
+include("reduction_cuda_utils.jl")
+
+@testset "test distributed cuda reduction op on surface of sphere" begin
+    FT = Float64
+    context = ClimaComms.context()
+    context_cpu = ClimaComms.context(ClimaComms.CPUDevice()) # CPU context for comparison
+    ClimaComms.init(context)
+    ClimaComms.init(context_cpu)
+
+    logger_stream = ClimaComms.iamroot(context) ? stderr : devnull
+    prev_logger = global_logger(ConsoleLogger(logger_stream, Logging.Info))
+    # Set up discretization
+    ne = 72
+    Nq = 4
+    ndof = ne * ne * 6 * Nq * Nq
+    @info "Configuration" device = context.device nprocs =
+        ClimaComms.nprocs(context) Ne = ne Nq = Nq ndof = ndof FT = FT
+    R = FT(6.37122e6) # radius of earth
+    domain = Domains.SphereDomain(R)
+    mesh = Meshes.EquiangularCubedSphere(domain, ne)
+    quad = Spaces.Quadratures.GLL{Nq}()
+    grid_topology = Topologies.Topology2D(context, mesh)
```

Comment:
Tests define different topologies depending on context, then check that functions return approx identical values on cpu / gpu devices; seems reasonable to me!

Are tests for `spacefillingcurve` implicit in this setup? For instance, in ClimaAtmos the topology is constructed with 

```
 topology = Topologies.Topology2D(
            comms_ctx,
            mesh,
            Topologies.spacefillingcurve(mesh),
        )
```

Reply:
Resolving. This is considered in `118` below.

Reply:
For `ClimaCore.jl`, `linearindices` are the default. 
https://github.com/CliMA/ClimaCore.jl/blob/main/src/Topologies/topology2d.jl#L187

Reply:
The results of the tests should be independent of the partitioning strategy. Some of the tests compare reduced values with analytical estimates.

Reply:
Noted thank you!